content/docs/cucumber/state.md

```diff
@@ -238,6 +238,211 @@ compile group: 'io.cucumber', name: 'cucumber-needle', version: '{{% version "cu
 
 There is no documentation yet, but the code is on [GitHub](https://github.com/cucumber/cucumber-jvm/tree/master/needle).
 
+# Details of using DI (e.g. using Guice)
+
+(since Cucumber 5)
+
+It is typically very simple to use Cucumber with a dependency injection framework.
+All that's needed is to add the appropriate dependency to the classpath.
+
+For example, if you want to use [Google Guice](https://github.com/google/guice) as DI
+simply add the `cucumber-guice` dependency to your pom.xml:
+
+```xml
+<dependencies>
+  [...]
+    <dependency>
+        <groupId>io.cucumber</groupId>
+        <artifactId>cucumber-guice</artifactId>
+        <version>${cucumber.version}</version>
+        <scope>test</scope>
+    </dependency>
+  [...]
+</dependencies>
+```
+
+Now all your step definitions, hooks, transformers, etc. will be created and supplied by a Guice injector.
```

Comment:
Does this happen automatically? From the text below it seems there is other configuration needed.